# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/nsHostResolver.cpp
# Commit: 26c5be98ea21
# Full Hash: 26c5be98ea21c8c331cefe9649e3eb53197cd821
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-08-26 03:24:37
# Regressor Bug: 1652667
# File Overlap Count: 1
# Description:
#   Bug 1652667 - P2: Telemetry for whether a connection is succeeded by using an HTTPSSVC record r=valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D87857
# ==============================================================================

diff -r 96bd98319c62 -r 26c5be98ea21 netwerk/dns/nsHostResolver.cpp
--- a/netwerk/dns/nsHostResolver.cpp	Tue Aug 25 17:29:32 2020 +0000
+++ b/netwerk/dns/nsHostResolver.cpp	Tue Aug 25 17:29:31 2020 +0000
@@ -522,7 +522,8 @@
     : nsHostRecord(key),
       DNSHTTPSSVCRecordBase(key.host),
       mTrrLock("TypeHostRecord.mTrrLock"),
-      mResultsLock("TypeHostRecord.mResultsLock") {}
+      mResultsLock("TypeHostRecord.mResultsLock"),
+      mAllRecordsExcluded(false) {}
 
 TypeHostRecord::~TypeHostRecord() { mCallbacks.clear(); }
 
@@ -614,8 +615,8 @@
   }
 
   auto& results = mResults.as<TypeRecordHTTPSSVC>();
-  nsCOMPtr<nsISVCBRecord> result =
-      GetServiceModeRecordInternal(aNoHttp2, aNoHttp3, results);
+  nsCOMPtr<nsISVCBRecord> result = GetServiceModeRecordInternal(
+      aNoHttp2, aNoHttp3, results, mAllRecordsExcluded);
   if (!result) {
     return NS_ERROR_NOT_AVAILABLE;
   }
@@ -637,6 +638,18 @@
   return NS_OK;
 }
 
+NS_IMETHODIMP
+TypeHostRecord::GetAllRecordsExcluded(bool* aResult) {
+  NS_ENSURE_ARG(aResult);
+
+  if (!mResults.is<TypeRecordHTTPSSVC>()) {
+    return NS_ERROR_NOT_AVAILABLE;
+  }
+
+  *aResult = mAllRecordsExcluded;
+  return NS_OK;
+}
+
 //----------------------------------------------------------------------------
 
 static const char kPrefGetTtl[] = "network.dns.get-ttl";